lib/clients/public.js

```diff
@@ -58,17 +58,28 @@ class PublicClient {
       } catch (e) {
         data = null;
       }
-      if (typeof callback === 'function') {
-        callback(err, response, data);
-        return;
-      }
+
       if (err) {
         err.response = response;
-        reject(err);
-      }
-      if (data === null) {
+      } else if (response.statusCode > 299) {
+        err = new Error(`HTTP ${response.statusCode} Error`);
```

Comment:
Use the `message` from the response here?

Reply:
Do you mean the `statusMessage` from the `response`, or the `message` from the `body`, or both? I'm inclined to try both.

Reply:
`message` from the body. `statusMessage` wouldn't be too bad either 🤷‍♂️